server/schedulers/hot_region.go

```diff
@@ -868,39 +868,68 @@ func (bs *balanceSolver) checkDstByPriorityAndTolerance(maxLoad, expect *storeLo
 // calcProgressiveRank calculates `bs.cur.progressiveRank`.
 // See the comments of `solution.progressiveRank` for more about progressive rank.
 func (bs *balanceSolver) calcProgressiveRank() {
-	srcLd := bs.stLoadDetail[bs.cur.srcStoreID].LoadPred.min()
-	dstLd := bs.stLoadDetail[bs.cur.dstStoreID].LoadPred.max()
+	src := bs.stLoadDetail[bs.cur.srcStoreID].LoadPred
+	dst := bs.stLoadDetail[bs.cur.dstStoreID].LoadPred
+	bs.cur.progressiveRank = 0
+	srcLd := src.min()
+	dstLd := dst.max()
 	peer := bs.cur.srcPeerStat
-	rank := int64(0)
+
 	if bs.rwTy == write && bs.opTy == transferLeader {
 		// In this condition, CPU usage is the matter.
 		// Only consider key rate or query rate.
```

Comment:
Does the comment need to change?

Reply:
Generally speaking, we would like to use qps or key as the dimension of qps instead of byte. or is it better to prompt the user when he/she sets it?

Reply:
Here you only have a variable named firstPriority. It's hard to get any information from it. If you think we need to make it clear, prefer to move this comment to the place where the value is set.
BTW, how about using a function for `bs.rwTy == write && bs.opTy == transferLeader`? There are lots of places using it.